cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -489,20 +490,28 @@ public List<ApplicationId> getAllAppVersionsAppIds(ApplicationReference appRef)
    * @throws IOException if failed to read metadata
    */
   public Map<ApplicationId, ApplicationMeta> getApplicationsForAppIds(
-      Collection<ApplicationId> appIds, Function<ApplicationId, SourceControlMeta> getSourceControlMeta)
+      Collection<ApplicationId> appIds,
+      FunctionWithException<ApplicationId, SourceControlMeta, IOException> getSourceControlMeta)
       throws IOException {
     List<List<Field<?>>> multiKeys = appIds.stream()
         .map(this::getApplicationPrimaryKeys)
         .collect(Collectors.toList());
-    SourceControlMeta sourceControlMeta = getSourceControlMeta.apply(appIds.iterator().next());
     return getApplicationSpecificationTable().multiRead(multiKeys)
         .stream()
         .collect(Collectors.toMap(
             AppMetadataStore::getApplicationIdFromRow,
-            row -> getSourceControlMetaAndDecodeRow(row, getSourceControlMeta)));
+            row -> {
+              try {
+                return getSourceControlMetaAndDecodeRow(row, getSourceControlMeta);
+              } catch (IOException e) {
+                throw new RuntimeException(e);
```

Comment:
In general, changing a checked exception into a runtime exception should be done infrequently, as it makes error handling difficult for the caller. 

In this particular case, getApplicationsForAppIds already throws an IOException, so it's better if this IOException just gets thrown directly. Probably better to avoid stream() and collect() in this scenario.

Reply:
Sure. Made the change